pkg/containers/containers.go

```diff
@@ -88,63 +90,64 @@ func (c *Containers) GetCgroupV1HID() int {
 
 // Populate populates Containers struct by reading mounted proc and cgroups fs.
 func (c *Containers) Populate() error {
-	err := c.procMountsCgroups()
-	if err != nil {
-		return err
+	if c.cgroupMP == "" {
+		err := c.findCgroupMounts()
+		if err != nil {
+			return err
+		}
 	}
 
 	return c.populate()
 }
 
 // getCgroupV1SSID finds cgroup v1 hierarchy ID.
-func initCgroupV1() error {
-	cgroupsFile := "/proc/cgroups"
-	file, err := os.Open(cgroupsFile)
+func (c *Containers) initCgroupV1() error {
+	c.cgroupV1 = true
+	hierarchyId, err := getCgroupV1HierarchyId()
 	if err != nil {
-		return fmt.Errorf("could not open cgroups file %s: %w", cgroupsFile, err)
+		return err
 	}
-	defer file.Close()
+	cgroupV1HierarchyID = hierarchyId
 
-	scanner := bufio.NewScanner(file)
-	for i := 1; scanner.Scan(); i++ {
-		sline := strings.Fields(scanner.Text())
-		if len(sline) < 2 || sline[0] != cgroupV1Controller {
-			continue
+	inContainer, err := runsOnContainerV1()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "containers: failed to detect if running on a cgroupv1 container: %s", err)
+	}
+
+	if inContainer {
+		os.Mkdir("./cpuset", 0644)
+		mountPath, err := filepath.Abs("./cpuset")
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to get absolute path of cputset mountpoint %s\n", mountPath)
+			return nil
 		}
-		intID, err := strconv.Atoi(sline[1])
-		if err != nil || intID < 0 {
-			return fmt.Errorf("error parsing %s: %w", cgroupsFile, err)
+
+		err = mountCpuset(mountPath)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "containers: failed to mount cgroupv1 controller: %s\n", err)
```

Comment:
if you continue and set `c.cgroupMP = mountPath` below althogh `err != nil` here it will be invalid

Reply:
You're right i've added a return above inside the if.